### PR TITLE
feat: add a `-workflow-input` option

### PR DESCRIPTION
**Pull request by laurentsimon**
See https://github.com/slsa-framework/slsa-verifier/issues/213

This PR uses `-workflow-input key=value` and the option can be passed multiple times.

### CODE REVIEW

File: verifiers/internal/gha/provenance.go [hunk 1]
```diff
@@ -224,6 +224,61 @@ func VerifyProvenance(env *dsselib.Envelope, provenanceOpts *options.ProvenanceO
...
+	}
+
+	// Verify all inputs.
+	for k, v := range inputs {
+		value, err := getAsString(pyldInputs, k)
+		if err != nil {
+			return fmt.Errorf("%w: cannot retrieve value of '%s'", serrors.ErrorMismatchWorkflowInputs, k)
+		}
+
+		if v != value {
```
Comment (asraa) [inline verifiers/internal/gha/provenance.go:276]: do you want a case-insensitive comparison here?

File: cli/slsa-verifier/main.go [hunk 2]
```diff
@@ -42,6 +66,9 @@ func main() {
 		"[optional] expected version the binary was compiled from. Uses semantic version to match the tag")
 	flag.BoolVar(&printProvenance, "print-provenance", false,
 		"print the verified provenance to std out")
+	inputs.kv = make(map[string]string)
+	flag.Var(&inputs, "workflow-input",
+		"[optional] a workflow input provided by a user at trigger time (e.g., workflow_dispatch) in the format 'key=value'.")
```
Comment (asraa) [inline cli/slsa-verifier/main.go:71]: This only supports workflow_dispatch, so e.g. should be clear that this is the only supported case

File: cli/slsa-verifier/main.go [hunk 3]
```diff
@@ -21,13 +27,31 @@ var (
...
+	inputs          workflowInputs
 	printProvenance bool
 )
 
 func experimentalEnabled() bool {
 	return os.Getenv("SLSA_VERIFIER_EXPERIMENTAL") == "1"
 }
 
+func (i *workflowInputs) String() string {
+	return "some representation"
```
Comment (asraa) [inline cli/slsa-verifier/main.go:39]: `fmt.Sprint` or `json.Marshal` could work here

File: cli/slsa-verifier/main.go (hunk 2 above)
Comment (laurentsimon) [reply cli/slsa-verifier/main.go]: Done, PTAL

File: cli/slsa-verifier/main.go (hunk 3 above)
Comment (laurentsimon) [reply cli/slsa-verifier/main.go]: done with `Sprintf`

**Comment by laurentsimon** [comment]
fyi @elefeint we're going to cut a release this week. We'll be able to update your project's documentation `-workflow-input release-version=x.y.z`. I'll send a PR.

**Pull request merged**